src/tests/components/ActivityForm.test.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import ActivityForm from '../../components/ActivityForm';
+import testState from '../fixtures/state';
+import Input from '../../components/Input';
+import shouldRenderErrorOnBlur from '../fixtures/shouldRenderErrorOnBlur';
+
+describe('ActivityForm', () => {
+  it('should render ActivityForm correctly', () => {
+    const wrapper = shallow(<ActivityForm onSubmit={() => {}} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should render ActivityForm with testState', () => {
+    const wrapper = shallow(
+      <ActivityForm on onSubmit={() => {}} activity={testState.activities.items[1]} />,
+    );
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should set errorMsg in inputs after invalid form submition', () => {
+    const wrapper = shallow(<ActivityForm onSubmit={() => {}} />);
+    expect(wrapper).toMatchSnapshot();
+    wrapper.find('form').simulate('submit', {
+      preventDefault: () => {},
+    });
+    expect(wrapper.state('classNoError').length).toBeGreaterThan(0);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should set name on input change', () => {
+    const value = 'New name';
+    const wrapper = shallow(<ActivityForm on onSubmit={() => {}} />);
+    wrapper
+      .find(Input)
+      .at(0)
+      .simulate('change', {
+        target: { value },
+      });
+    expect(wrapper.state('name')).toBe(value);
+  });
+
+  it('should set teacher on input change', () => {
+    const value = 'New teacher';
+    const wrapper = shallow(<ActivityForm on onSubmit={() => {}} />);
+    wrapper
+      .find(Input)
+      .at(1)
+      .simulate('change', {
+        target: { value },
+      });
+    expect(wrapper.state('teacher')).toBe(value);
+  });
+
+  it('should set room on input change', () => {
+    const value = 'New room';
+    const wrapper = shallow(<ActivityForm on onSubmit={() => {}} />);
+    wrapper
+      .find(Input)
+      .at(2)
+      .simulate('change', {
+        target: { value },
+      });
+    expect(wrapper.state('room')).toBe(value);
+  });
+
+  it('should set classNo on input change', () => {
+    const value = '123';
+    const wrapper = shallow(<ActivityForm on onSubmit={() => {}} />);
+    wrapper
+      .find(Input)
+      .at(3)
+      .simulate('change', {
+        target: { value },
+      });
+    expect(wrapper.state('classNo')).toBe(value);
```

Comment:
in my opinion, you should test fullfiling all 3 inputs at once and check if submit is invoked with proper args

Reply:
ok... I see that you are doing this. 
You don't need to test if state is changed properly.